### DISCUSSION TITLE
Why MiniExcel?

### DISCUSSION THREAD

**Discussion by shps951023** [mini-software/MiniExcel#412] [Ideas]
Why MiniExcel, and what business scenarios you solved by MiniExcel? What features are you looking forward to adding in the future?

為何選擇 MiniExcel，解決什麼業務需求? 最期待未來增加什麼功能?

为何选择 MiniExcel，解决什么业务需求? 最期待未来增加什么功能?

Really appreciate!

**Comment by andy505050** [comment]
為何選擇 MiniExcel ?
1. 記憶體用量小
2. API 比 Epplus 和 NPOI 好用很多
3. Checkmark 不會掃出令人無言的問題

最期待未來增加什麼功能?
1. 期待增加可以用程式控制儲存格樣式
2. 希望有 MiniWord !!

**Reply by chkuo2** [reply]
support ods import and export

**Comment by SirParser** [comment]
Would it be possible/difficult to add Python(pandas)-like functionality for existing Excel file, using DataReader as input? 
_pandas.DataFrame.to_excel (..., columns=None, header=True'False or array of strings, ..., startrow=0, startcol=0, ...)_
where header = array of column names (or false if none or true if default), 
startrow/startcol - where the dump should start, e.g in cell B5, without overwriting existing data.
The key point is to keep the performance benefits of DataReader, and allow dumping the data to existing file.

**Comment by sgf** [comment]
能不能开放 discussions 的创建?
我有一些使用疑问,例如:使用Query的时候怎么设置停止读取.
startCell 是开始.但是读到哪里为止呢?
似乎没有可以设置停止读取的机制或者参数,建议增加.
比如:如果遇到不是想要的数据类型,就停止读取.
比如:后续没有数据了,是最后一行就停止读取.
比如:值是指定的值,就停止读取.
等等条件.
这个条件最好是做成表达式.
